recipes/mysql-connector-cpp/all/conanfile.py

```diff
@@ -0,0 +1,171 @@
+
+requird_conan_version = ">=1.54.0"
+
+class MysqlConnectorCPPRecipe(ConanFile):
+    name = "mysql-connector-cpp"
+    license = "GPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://dev.mysql.com/doc/connector-cpp/8.1/en/"
+    description = "A MySQL database connector for C++ applications that connect to MySQL servers"
+    topics = ("mysql", "connector", "libmysqlclient", "jdbc")
+    package_type = "library"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_jdbc": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "with_jdbc": True}
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "9" if Version(self.version) >= "8.0.27" else "5.3",
+            "clang": "6",
+        }
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("protobuf/3.21.12")
+
+    def requirements(self):
+        self.requires("zlib/[>=1.2.10 <2]")
+        self.requires("lz4/1.9.4")
+        self.requires("zstd/1.5.5")
+        self.requires("protobuf/3.21.12")
+        self.requires("boost/1.83.0")
+        self.requires("openssl/[>=1.1.1 <4]")
+        self.requires("libmysqlclient/8.1.0")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self.source_folder)
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 192)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.")
+
+        # I dont have a windows computer to test it
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support windows for now")
+
+        # Sice 8.0.17 this doesn't support shared library on MacOS.
+        # https://github.com/mysql/mysql-server/blob/mysql-8.0.17/cmake/libutils.cmake#L333-L335
+        if self.settings.compiler == "apple-clang" and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support shared library on apple-clang")
+
+        # mysql < 8.0.29 uses `requires` in source code. It is the reserved keyword in C++20.
+        # https://github.com/mysql/mysql-server/blob/mysql-8.0.0/include/mysql/components/services/dynamic_loader.h#L270
+        if self.settings.compiler.get_safe("cppstd") == "20" and Version(self.version) < "8.0.29":
```

Comment:
```suggestion
        if "20" in self.settings.compiler.get_safe("cppstd") and Version(self.version) < "8.0.29":
```

Because `gnu20` is a valid cppstd value.